src/libasr/codegen/asr_to_cpp.cpp

```diff
@@ -695,45 +680,30 @@ Kokkos::View<T*> from_std_vector(const std::vector<T> &v)
         if (x.m_type->type == ASR::ttypeType::Integer) {
             if (x.m_op == ASR::unaryopType::UAdd) {
                 // src = src;
-                last_unary_plus = false;
-                return;
             } else if (x.m_op == ASR::unaryopType::USub) {
                 src = "-" + src;
-                last_unary_plus = true;
-                last_binary_plus = false;
             } else if (x.m_op == ASR::unaryopType::Invert) {
                 src = "~" + src;
-                last_unary_plus = false;
-                last_binary_plus = false;
             } else if (x.m_op == ASR::unaryopType::Not) {
                 src = "!" + src;
-                last_unary_plus = false;
-                last_binary_plus = false;
             } else {
                 throw CodeGenError("Unary type not implemented yet for Integer");
             }
             return;
         } else if (x.m_type->type == ASR::ttypeType::Real) {
             if (x.m_op == ASR::unaryopType::UAdd) {
                 // src = src;
-                last_unary_plus = false;
             } else if (x.m_op == ASR::unaryopType::USub) {
                 src = "-" + src;
-                last_unary_plus = true;
-                last_binary_plus = false;
             } else if (x.m_op == ASR::unaryopType::Not) {
                 src = "!" + src;
-                last_unary_plus = false;
-                last_binary_plus = false;
             } else {
                 throw CodeGenError("Unary type not implemented yet for Real");
             }
             return;
         } else if (x.m_type->type == ASR::ttypeType::Logical) {
             if (x.m_op == ASR::unaryopType::Not) {
                 src = "!" + src;
-                last_unary_plus = false;
-                last_binary_plus = false;
```

Comment:
Let's assign to `last_expr_precedence` in this function or add a comment why it is not needed.